diff --git a/src/Makefile.am b/src/Makefile.am
index 2aa2221..dc9240d 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -210,7 +210,3 @@ split_ncvars.pl: split_ncvars/split_ncvars.pl.in Makefile
 	$(do_subst) < $(srcdir)/split_ncvars/split_ncvars.pl.in > split_ncvars.pl
 	chmod +x split_ncvars.pl
 
-install-exec-hook:
-	ln $(DESTDIR)$(bindir)/split_ncvars.pl $(DESTDIR)$(bindir)/split_ncvars
-install-data-hook:
-	ln $(DESTDIR)$(mandir)/man1/split_ncvars.pl.1 $(DESTDIR)$(mandir)/man1/split_ncvars.1
