diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac452ab4..eb39d743 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -105,6 +105,7 @@ find_package(sigio 2.3.0 REQUIRED)
 find_package(sp 2.3.3 REQUIRED)
 find_package(ip 3.3.3 REQUIRED)
 find_package(g2 3.4.3 REQUIRED)
+find_package(g2c REQUIRED)
 find_package(sigio 2.3.0 REQUIRED)
 
 # If doxygen documentation we enabled, build it. This must come before
